api/views/__init__.py

```diff
@@ -153,16 +154,19 @@ class DomainAddressViewSet(SaveToRequestUser, viewsets.ModelViewSet):
     def get_queryset(self):
         return DomainAddress.objects.filter(user=self.request.user)
 
-    def perform_create(self, serializer):
+    def perform_create(self, serializer: BaseSerializer[DomainAddress]) -> None:
         try:
             serializer.save(user=self.request.user)
         except IntegrityError:
+            assert isinstance(self.request.user, User)
             domain_address = DomainAddress.objects.filter(
                 user=self.request.user, address=serializer.validated_data.get("address")
             ).first()
-            raise ConflictError(
-                {"id": domain_address.id, "full_address": domain_address.full_address}
-            )
+            existing_id = existing_address = None
+            if domain_address:
+                existing_id = domain_address.id
+                existing_address = domain_address.full_address
```

Comment:
(suggestion, blocking) If there is an `IntegrityError` that is not from conflicting DomainAddress, I think we should just raise the `IntegrityError` instead. From reading `IntegrityError` [description on the Django doc](https://code.djangoproject.com/wiki/IntegrityError), a good test case would be trying to create a `DomainAddress` with a user ID that does not exist.

Reply:
I don't think that is possible through this API endpoint. We first check that the user is authorized, and if the user doesn't exist, we'd fail way before we get to creating a new `DomainAddress`.

I agree we should be raising the `IntegrityError`, which means the duplicate check needs to be somewhere else.